src/background.js

```diff
@@ -1,106 +1,57 @@
-    this.prefs = prefs // shorthand language strings ['au', 'de']
-    this.langs = languages // language strings ['en-au', 'de-de']
-    this.ownWords = ownWords // word strings ['kablam', 'shebang']
-    this.spells = this.createSpellCheckers() // nspell instances
-  }
-
-  // create spell checkers in order of languages specified by user
-  createSpellCheckers () {
-    if (this.prefs.length !== this.dicts.length) {
-      console.warn('MultiDict: Language prefs and user dictionary length not equal. Aborting.')
-      return
-    }
-
-    const spells = {}
-    for (let i = 0; i < this.dicts.length; i++) {
-      spells[this.prefs[i]] = nspell(this.dicts[i])
-    }
-    return spells
-  }
-
-  // add word to user.ownWords and update existing spell checker instances
-  addWord (word) {
-    user.ownWords.push(word)
-    Object.keys(this.spells).forEach((language) => { this.spells[language].add(word) })
-  }
-
-  // remove word from user.ownWords and update existing spell checker instances
-  removeWord (word) {
-    user.ownWords.remove(word)
-    Object.keys(this.spells).forEach((language) => { this.spells[language].remove(word) })
-  }
-}
-
-// expected: { type: '', detectedLanguage: '', content: '' }
-// checkSpelling returns: { cleanedContent: [], misspeltWords: [], suggestions: {}}
-function check (message) {
-  switch (message.detectedLanguage) {
-    case 'unreliable':
-      messageHandler.postMessage({
-        type: 'spelling',
-        suggestions: checkSpelling(user.spells[user.prefs[0]], message.content)
-      })
-      break
-
-    default:
-      for (const pref of user.prefs) {
-        if (user.langs.includes(`${message.detectedLanguage}-${pref}`)) {
-          messageHandler.postMessage({
-            type: 'spelling',
-            spelling: checkSpelling(user.spells[pref], message.content)
-          })
-          break
-        }
-      }
-  }
-}
-
 // saves a word in browser storage and calls user.addWord
 async function addCustomWord (word) {
-  if (customWords.includes(word)) {
+  if (!word || customWords.includes(word)) {
     return
   }
 
-  console.log('saving word: ' + word)
-
-  customWords.push(word)
-  const error = await browser.storage.sync.set({ personal: customWords })
+  const error = await browser.storage.sync.set({ personal: [...customWords, word] })
 
   if (error) {
-    customWords.pop() // remove the last word we added to customWords if we fail to save it
     console.error(`MultiDict: Adding to personal dictionary failed with error: ${error}`)
   } else {
+    customWords.push(word)
     user.addWord(word)
   }
+
+  console.log('add result', customWords)
 }
 
 // removes a word from browser storage and calls user.removeWord
 async function removeCustomWord (word) {
-  if (!customWords.includes(word)) {
+  if (!word || !customWords.includes(word)) {
     return
   }
 
-  console.log('deleting word :' + word)
+  if (word === 'ClearAll') {
```

Comment:
Should make sure to remove this after adding feature to clear personal dictionary